caffe2/operators/utility_ops.cc

```diff
@@ -833,6 +833,26 @@ OPERATOR_SCHEMA(NanCheck)
         "Tensor to copy input into if no NaNs or inf."
         " Can be in-place");
 
+template <>
+bool IsNanOp<CPUContext>::RunOnDevice() {
+  auto& X = Input(0);
+  auto* Y = Output(0, X.sizes(), at::dtype<uint8_t>());
+  const float* X_data = X.template data<float>();
```

Comment:
We should assert it's `float` type. Cannot direct convert without checking.

Another option: create a template for data type as well. How to do that, you can check other op's implementation. :-)